internal/pkg/warewulfd/api/api.go

```diff
@@ -0,0 +1,43 @@
+package api
```

Comment:
I don't object to breaking the api code into separate files; but I question whether the api should actually be a separate package from the warewulfd package. What would you think of keeping both of these in the warewulfd package, but splitting it into `api.go`, `container_api.go`, `node_api.go`, `profile_api.go`, and `overlay_api.go`?

I'm far from versed in best practices here; so if you can point me to some guidelines for where to split golang into separate packages, I'd love to read it.

Reply:
I might be wrong about this; so don't hesitate to push back. As long as it's only `warewulfd` that imports `api`, my fears about a generic `api` package are probably unfounded.